authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/ChainAuthorizationProperties.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.authorization.ranger;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * ChainAuthorizationProperties is a class that implements {@link AuthorizationProperties}
+ * interface. <br>
+ * <br>
+ * Configuration Example: <br>
+ * "authorization.chain.plugins" = "hive1,hdfs1" <br>
+ * "authorization.chain.hive1.provider" = "ranger"; <br>
+ * "authorization.chain.hive1.catalog-provider" = "hive"; <br>
+ * "authorization.chain.hive1.ranger.auth.type" = "simple"; <br>
+ * "authorization.chain.hive1.ranger.admin.url" = "http://localhost:6080"; <br>
+ * "authorization.chain.hive1.ranger.username" = "admin"; <br>
+ * "authorization.chain.hive1.ranger.password" = "admin"; <br>
+ * "authorization.chain.hive1.ranger.service.name" = "hiveDev"; <br>
+ * "authorization.chain.hdfs1.provider" = "ranger"; <br>
+ * "authorization.chain.hdfs1.catalog-provider" = "hadoop"; <br>
+ * "authorization.chain.hdfs1.ranger.auth.type" = "simple"; <br>
+ * "authorization.chain.hdfs1.ranger.admin.url" = "http://localhost:6080"; <br>
+ * "authorization.chain.hdfs1.ranger.username" = "admin"; <br>
+ * "authorization.chain.hdfs1.ranger.password" = "admin"; <br>
+ * "authorization.chain.hdfs1.ranger.service.name" = "hdfsDev"; <br>
+ */
+public class ChainAuthorizationProperties implements AuthorizationProperties {
+  public static final String PLUGINS_SPLITTER = ",";
+  /** Chain authorization plugin names */
+  public static final String CHAIN_PLUGINS_PROPERTIES_KEY = "authorization.chain.plugins";
+
+  /** Chain authorization plugin provider */
+  public static final String CHAIN_CATALOG_PROVIDER = "authorization.chain.*.catalog-provider";
+
+  /** Chain authorization plugin provider */
+  public static final String CHAIN_PROVIDER = "authorization.chain.*.provider";
+
+  static Map<String, String> fetchAuthPluginProperties(
+      String pluginName, Map<String, String> properties) {
```

Comment:
I split this PR from https://github.com/apache/gravitino/pull/5786, Because https://github.com/apache/gravitino/pull/5786 very big.
I need `fetchAuthPluginProperties()` function in the int the #5786, and I had add `fetchAuthPluginProperties()` unit test in the here.
- https://github.com/apache/gravitino/pull/5791/files#diff-e4c29f3dc548a012bc5039ee8d4d0d3e1f5bdf08a2edd908cd3e7dcc74445f4bR181-R212

Reply:
I mean within this function, not outside the function.